File: src/main/java/cz/vhromada/catalog/mapper/SongMapper.java
package cz.vhromada.catalog.mapper;

import cz.vhromada.catalog.entity.Song;

import org.mapstruct.Mapper;

/**
 * An interface represents mapper for song.
 *
 * @author Vladimir Hromada
 */
@Mapper
public interface SongMapper {

    /**
     * Maps entity song to domain song.
     *
     * @param source entity song
     * @return mapped domain song
     */
    cz.vhromada.catalog.domain.Song map(Song source);

    /**
     * Maps domain song to entity song.
     *
     * @param source domain song
     * @return mapped entity song
     */
    Song mapBack(cz.vhromada.catalog.domain.Song source);

}

File: src/main/kotlin/cz/vhromada/catalog/mapper/SongMapper.kt
package cz.vhromada.catalog.mapper

import cz.vhromada.catalog.entity.Song
import cz.vhromada.common.mapper.Mapper
import org.springframework.stereotype.Component

/**
 * A class represents mapper for song.
 *
 * @author Vladimir Hromada
 */
@Component("songMapper")
class SongMapper : Mapper<Song, cz.vhromada.catalog.domain.Song> {

    override fun map(source: Song): cz.vhromada.catalog.domain.Song {
        return cz.vhromada.catalog.domain.Song(
                id = source.id,
                name = source.name!!,
                length = source.length!!,
                note = source.note,
                position = source.position)
    }

    override fun mapBack(source: cz.vhromada.catalog.domain.Song): Song {
        return Song(
                id = source.id,
                name = source.name,
                length = source.length,
                note = source.note,
                position = source.position)
    }

}
